thrift/arg2/kv_iterator.go

```diff
@@ -30,14 +30,10 @@ func NewKeyValIterator(arg2Payload []byte) (KeyValIterator, error) {
 		return KeyValIterator{}, io.EOF
 	}
 
-	// We don't hold on to rbuf to avoid the associated alloc
-	rbuf := typed.NewReadBuffer(arg2Payload)
-	leftPairCount := rbuf.ReadUint16()
-
+	leftPairCount := binary.BigEndian.Uint16(arg2Payload[0:2])
```

Comment:
since we're not doing anything else with `leftPairCount`, maybe just inline below?

Reply:
Since we're also casting to `int`, and we do the `Next` in that line, thought separate lines would be easier so there's not too much cognitive overhead in that expression.